refl1d/magnetism.py

```diff
@@ -474,13 +474,24 @@ def parvec(vector, name, limits):
             mbelow, name=name + " mbelow", limits=(-inf, inf))
         self.mabove = Parameter.default(
             mabove, name=name + " mabove", limits=(-inf, inf))
+
+        # if only tabove or tbelow is defined then they are made equal
+        # this is to deal with the situation of a non-magnetic
+        # layer next to a magnetic one
+        if tabove is None and tbelow is None:
+            tbelow = tabove = DEFAULT_THETA_M
+        elif tabove is not None and tbelow is None:
+            tbelow = tabove
+        elif tbelow is not None and tabove is None:
+            tabove = tbelow
+
         self.tbelow = Parameter.default(
             tbelow, name=name + " tbelow", limits=(0, 360))
         self.tabove = Parameter.default(
             tabove, name=name + " tabove", limits=(0, 360))
         if len(self.dz) != len(self.drhoM):
             raise ValueError("Need one dz for each drhoM")
-        if len(self.dthetaM) > 0 and len(self.drhoM) != len(self.dthetaM):
+        if 0 < len(self.dthetaM) != len(self.drhoM):
```

Comment:
I think this was a result of my linter picking it up. I am happy to leave it as it was.

Reply:
I'm surprised the linter complained. Unless it was complaining about the `len(a) > 0` clause?

Unlike lists, if `obj` is a numpy array then `if obj` raises an error rather than checking length so I habitually use `len(a) > 0` and ignore the linter.

I'm not averse to using new syntax, but a small amount of unscientific polling says the `0 < a != b` construct is harder to read.